apps/pay/app/setuppwa/page.tsx

```diff
@@ -1,51 +1,53 @@
-import { useRouter } from "next/router"
-import React from "react"
+"use client"
+import { useRouter } from "next/navigation"
+import React, { useEffect } from "react"
 
 import CurrencyDropdown from "../../components/Currency/currency-dropdown"
 
 const SetupPwa = () => {
   const router = useRouter()
   const [username, setUsername] = React.useState<string>("")
-  const username_from_local = localStorage.getItem("username")
-  const display_currency_from_local = localStorage.getItem("display")
+
+  let usernameFromLocal: string | null = null
+  let displayCurrencyFromLocal: string | null = null
+
+  useEffect(() => {
+    usernameFromLocal = localStorage.getItem("username")
+    displayCurrencyFromLocal = localStorage.getItem("display")
+  }, [])
+
   const [selectedDisplayCurrency, setSelectedDisplayCurrency] = React.useState("USD")
 
+  React.useEffect(() => {
+    if (usernameFromLocal) {
+      window.history.pushState(
+        {},
+        "",
+        `${usernameFromLocal}??display=${displayCurrencyFromLocal}`,
+      )
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
when we are using this comment we should explain in comment why we need that. it's typically not needed and a sign that the overall design of the component may be suboptimal when needed